src/command_modules/azure-cli-network/azure/cli/command_modules/network/dns_zone.py

```diff
@@ -0,0 +1,26 @@
+#---------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#---------------------------------------------------------------------------------------------
+
+"""
+Commands for DNS Zone Management
+"""
+
+#pylint: disable=too-many-arguments
+def create_dns_zone(client, resource_group_name, zone_name, location='global', tags=None,
+                    if_none_match=False):
+    body = {'location': location}
+    if tags:
+        body['tags'] = tags
+
+    kwargs = {
+        'resource_group_name':resource_group_name,
+        'zone_name': zone_name,
+        'parameters': body
+    }
+
+    if if_none_match:
+        kwargs['if_none_match'] = '*'
+
+    return client.create_or_update(**kwargs)
```

Comment:
Alternatively, you could skip the custom command entirely and simply reflect on the SDK method. That way you'd pick up any extra parameters for free (like if_match for example). That's the "danger" with custom commands is that as the SDK exposes more functionality, the CLI command becomes out of date until it's noticed.  With reflected commands, the new parameters are immediately exposed.

However, it is a little more work, so I am also fine with this as a custom command. There are examples of both throughout the CLI.